src/core/WakaTimeCore.ts

```diff
@@ -221,6 +228,17 @@ class WakaTimeCore {
     return items.loggingType;
   }
 
+  generateProjectFromDevSites(url: string): string | null {
+    const githubUrls = ['https://github.com', 'https://github.dev'];
+    for (const githubUrl of githubUrls) {
+      if (url.startsWith(githubUrl)) {
+        const newUrl = url.replace(githubUrl, '');
+        return newUrl.split('/')[2];
```

Comment:
Then this becomes:

`return newUrl.split('/')[1] || null;`

To return `null` instead of `undefined` when the url isn't a repo.